UPGRADING.md

```diff
@@ -0,0 +1,50 @@
+# 3.0.0 Migration Guide
+
+The v3.0.0 release of `google-cloud-bigtable` deprecates the previous `google.cloud.bigtable.Client` class in favor of distinct clients for the two API surfaces, supporting both sync and async calls:
+- Data API: 
+  - `google.cloud.bigtable.data.BigtableDataClient`
+  - `google.cloud.bigtable.data.BigtableDataClientAsync`
+- Admin API: 
+  - `google.cloud.bigtable.admin.BigtableInstanceAdminClient`
+  - `google.cloud.bigtable.admin.BigtableInstanceAdminAsyncClient`
+  - `google.cloud.bigtable.admin.BigtableTableAdminClient`
+  - `google.cloud.bigtable.admin.BigtableTableAdminAsyncClient`
+
+The deprecated client will remain available as an alternative API surface, which internally delegates calls to the respective new clients. For most users, existing code will continue to work as before. But there may be some breaking changes associated with this update, which are detailed in this document.
+
+### Admin Client Migration
```

Comment:
Currently it is more like a quick-start guide, instead of the migration guide.

For migration, we should start from what user have from legacy client and suggest what to do from there.

For example:
Path A (if you have time) - Add new import, remove old import, and convert all methods to new style, rebuild, test.
Path B (if you don't have time, but want to get new features) - Get new client via legacy client and use it. 

And indicate the path B is only for backward compatible phase and will be eventually deprecated.